### PR TITLE
BN055 IMU

### PR DESCRIPTION
**Pull request by neworderofjamie**
I haven't got round to finishing off the work on refactoring the Stone CX model I started at the hackathon, but this P.R. has the code for interfacing with BN055 IMU (there's a spare one on my desk if anyone wants it @tenxlenx). The current interface does not support all features of the device but, it would be trivial to port these from the [Arduino version](https://github.com/adafruit/Adafruit_BNO055) of someone required it.

### CODE REVIEW

**Comment by neworderofjamie** [comment]
Seems like casting modern enums is stricter on modern compilers than whatever my TX1 is running - will add ``static_cast`` left, right and centre

File: include/common/bn055_imu.h [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        LINEAR_ACCEL_DATA_Z_LSB_ADDR = 0X2C,
+        LINEAR_ACCEL_DATA_Z_MSB_ADDR = 0X2D,
+
+        // Gravity data registers
+        GRAVITY_DATA_X_LSB_ADDR = 0X2E,
+        GRAVITY_DATA_X_MSB_ADDR = 0X2F,
+        GRAVITY_DATA_Y_LSB_ADDR = 0X30,
+        GRAVITY_DATA_Y_MSB_ADDR = 0X31,
+        GRAVITY_DATA_Z_LSB_ADDR = 0X32,
+        GRAVITY_DATA_Z_MSB_ADDR = 0X33,
```
[the remark this answers is not in the document]
Comment (neworderofjamie) [reply include/common/bn055_imu.h]: They aren't but all these enums were copy-and-pasted from the Arduino interface (with a bit of modern c++ification by regex) so I'd rather leave them alone so it's clear what registers the hardware actually exposes

File: examples/bn055/bn055.cc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#include "common/logging.h"
+#include "common/bn055_imu.h"
+
+// Standard C includes
+#include <cmath>
+
+using namespace BoBRobotics;
+
+int main()
+{
```
[the remark this answers is not in the document]
Comment (neworderofjamie) [reply examples/bn055/bn055.cc]: Done

File: src/common/bn055_imu.cc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+namespace BoBRobotics
+{
+constexpr uint8_t BN055::imuID;
+//----------------------------------------------------------------------------
+BN055::BN055(OperationMode mode, const char *path, int slaveAddress)
+{
+    init(mode, path, slaveAddress);
+}
+//----------------------------------------------------------------------------
+void BN055::init(OperationMode mode, const char *path, int slaveAddress)
```
[the remark this answers is not in the document]
Comment (neworderofjamie) [reply src/common/bn055_imu.cc]: done

**Comment by neworderofjamie** [comment]
Implemented most of your suggestions and remembered the task I couldn't face doing before - a units interface (one of the nice things with this chip is that it returns 'real' units)

File: include/common/bn055_imu.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexdewar) [reply include/common/bn055_imu.h]: Fair enough

**Comment by alexdewar** [comment]
Cool -- all looks good to me. And the use of the units lib is a nice feature.

I've pushed a fix for an unrelated gcc warning so CI passes.

**Pull request merged**